src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -123,27 +123,35 @@
                                    (repeatedly children #(random-structured-fact (rand-nth (range depth))
                                                                                  (rand-nth (range children)))))})))
 
+(defn update-catalog [catalog stamp]
+  (assoc catalog "producer_timestamp" stamp))
+
 (defn maybe-tweak-catalog
   "Slightly tweak the given catalog, returning a new catalog, `rand-percentage`
-  percent of the time."
-  [catalog rand-percentage]
-  (if (< (rand 100) rand-percentage)
-    (rand-catalog-mutation catalog)
-    catalog))
-
-(defn update-catalog [catalog]
-  (assoc catalog "producer_timestamp" (time/now)))
+   percent of the time."
+  [catalog rand-percentage stamp]
+  (let [catalog' (update-catalog catalog stamp)]
+    (if (< (rand 100) rand-percentage)
+      (rand-catalog-mutation catalog')
+      catalog')))
+
+(defn jitter
+  "jitter a timestamp (rand n) milliseconds in the forward direction"
```

Comment:
I'm a little bit confused here. Comment says milliseconds but the code below says seconds. Also below is using rand, which returns a floating point number, but time/seconds is type hinted as an integer (and the java method it calls also only accepts an integer). There must be some auto casting magic going on. I think using rand-int here and above would be less confusing.